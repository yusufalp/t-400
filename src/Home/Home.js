import React from 'react';
import Start from '../Start/Start';

class Home extends React.Component {
  render() {
    return (
      <div>
        <h1>Welcome to T-400</h1>
        <h2>where we are going to test your limits and endurance. JK, this is just a game</h2>
        <p>if you get all 10 questions correct though, you will be called 'Terminator'... of this trivia game :)</p>
        <Start />
      </div>
    );
  }
}

export default Home;