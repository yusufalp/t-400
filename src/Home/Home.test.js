import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Home from './Home';
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe('Home component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Home />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Home title", () => {
    const wrapper = shallow(<Home />);
    const welcome = <h1>Welcome to T-400</h1>;
    expect(wrapper.contains(welcome)).toEqual(true);
  });

  it('renders one h1 element that is the title', () => {
    const wrapper = shallow(<Home />)
    const title = wrapper.find('h1')
    expect(title).toHaveLength(1)
  });

  it('renders button numbers', () => {
    const wrapper = shallow(<Home />)
    const button = wrapper.find('button')
    expect(button).toHaveLength(2)
  })
})