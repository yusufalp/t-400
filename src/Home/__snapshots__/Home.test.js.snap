// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders the UI as expected 1`] = `
<div
  className="title"
>
  <h1>
    Welcome to T-400
  </h1>
  <h2>
    where we are going to test your limits and endurance.
  </h2>
  <div
    className="content"
  >
    <p>
      JK, this is just a game
    </p>
    <p>
      if you get all 10 questions correct though, you will be called 'Terminator'... of this trivia game :)
    </p>
    <p>
      Ready to start?
    </p>
  </div>
  <button
    onClick={[Function]}
    type="submit"
  >
    Yes
  </button>
  <button
    onClick={[Function]}
    type="submit"
  >
    Absolutely Yes!
  </button>
</div>
`;
